features/kiesrecht.feature

```diff
@@ -108,46 +108,235 @@ Rule: Uitsluiting van Europees kiesrecht wordt alleen opgenomen wanneer de eindd
+
+    Scenario: einddatum uitgesluiting van kiesrecht ligt in de toekomst
+      Gegeven het systeem heeft een persoon met de volgende gegevens
+      | naam                | waarde    |
+      | burgerservicenummer | 999990001 |
+      En de persoon heeft de volgende 'kiesrecht' gegevens
+      | naam                                    | waarde |
+      | uitgeslotenVanKiesrecht                 | true   |
+      | einddatum uitsluiting kiesrecht (38.20) | morgen |
+      Als personen wordt gezocht met de volgende parameters
+      | naam                | waarde                          |
+      | type                | RaadpleegMetBurgerservicenummer |
+      | burgerservicenummer | 999990001                       |
+      | fields              | burgerservicenummer,kiesrecht   |
+      Dan heeft de response een persoon met alleen de volgende gegevens
+      | naam                | waarde    |
+      | burgerservicenummer | 999990001 |
+      En heeft de persoon alleen de volgende 'kiesrecht' gegevens
+      | naam                                | waarde |
+      | uitgeslotenVanKiesrecht             | true   |
+      | einddatumUitsluitingKiesrecht.type  | Datum  |
+      | einddatumUitsluitingKiesrecht.datum | morgen |
+
+    Abstract Scenario: einddatum uitsluiting van kiesrecht <subtitel>
+      Gegeven het systeem heeft een persoon met de volgende gegevens
+      | naam                | waarde    |
+      | burgerservicenummer | 999990001 |
+      En de persoon heeft de volgende 'kiesrecht' gegevens
+      | naam                                    | waarde      |
+      | uitgeslotenVanKiesrecht                 | true        |
+      | einddatum uitsluiting kiesrecht (38.20) | <gba datum> |
+      Als personen wordt gezocht met de volgende parameters
+      | naam                | waarde                          |
+      | type                | RaadpleegMetBurgerservicenummer |
+      | burgerservicenummer | 999990001                       |
+      | fields              | burgerservicenummer,kiesrecht   |
+      Dan heeft de response een persoon met alleen de volgende gegevens
+      | naam                | waarde    |
+      | burgerservicenummer | 999990001 |
+
+      Voorbeelden:
+      | sub titel             | gba datum |
+      | is gelijk aan vandaag | vandaag   |
+      | ligt in het verleden  | gisteren  |
+
+  Rule: wanneer alleen het jaar van de einddatum uitsluiting bekend is, dan wordt uitsluiting van kiesrecht opgenomen tot en met dat jaar
+
+    Scenario: einddatum uitsluiting van kiesrecht is dit jaar
+      Gegeven het systeem heeft een persoon met de volgende gegevens
+      | naam                | waarde    |
+      | burgerservicenummer | 999990001 |
+      En de persoon heeft de volgende 'kiesrecht' gegevens
+      | naam                                    | waarde   |
+      | uitgeslotenVanKiesrecht                 | true     |
+      | einddatum uitsluiting kiesrecht (38.20) | dit jaar |
```

Comment:
ik vraag me af of met "dit jaar" nog duidelijk is hoe die waarde eruit ziet. De logica voor wat echt getoond moet worden zit dan "verstopt" in de automation code. Het scenario wordt hiermee abstracter, meer een concreet voorbeeld met een echt uitgewerkt voorbeeld.

Is er geen manier om met gefixeerde datums te werken? In de Als stap, zoals in het scenario stond: "Als de persoon op 15 maart 2022 wordt geraadpleegd met de volgende parameters"
Dit staat ook zo in de leeftijd_bepaling feature. Werkt dat daar ook niet?

Reply:
> ik vraag me af of met "dit jaar" nog duidelijk is hoe die waarde eruit ziet. De logica voor wat echt getoond moet worden zit dan "verstopt" in de automation code. Het scenario wordt hiermee abstracter, meer een concreet voorbeeld met een echt uitgewerkt voorbeeld.
> 
> Is er geen manier om met gefixeerde datums te werken? In de Als stap, zoals in het scenario stond: "Als de persoon op 15 maart 2022 wordt geraadpleegd met de volgende parameters" Dit staat ook zo in de leeftijd_bepaling feature. Werkt dat daar ook niet?

Ja, ik kan het werkende krijgen, maar dan kan de scenario's niet meer worden gebruikt om de API in een productie-like omgeving te valideren. Om het raadplegen op een specifieke datum mogelijk te maken, moet er logica worden ge-mockt. In dit geval, dat de datum van vandaag gelijk is aan 15 maart 2022. En dan moet er ook logica worden toegevoegd om het mocken aan/uit te kunnen zetten. In productie wil je niet dat de mock wordt gebruikt.
Dus naar mijn mening niet een handige optie.
Kan er niet met extra documentatie worden uitgelegd wat met de datum termen wordt bedoeld? Verder kan ook met automation code en cucumber extra informatie in de rapportage worden toegevoegd om te laten zien wat het resultaat is van de stappen. Zie onderstaande screendumps voor voorbeelden van de extra informatie die worden toegevoegd:

![image](https://user-images.githubusercontent.com/18151729/164173028-ccc6c98b-1302-467e-8777-03056699272e.png)

![image](https://user-images.githubusercontent.com/18151729/164173214-0fb01f56-9e92-4545-8b67-c66a2e740ace.png)